e2e/__tests__/jest.config.ts.test.ts

```diff
@@ -74,50 +79,73 @@ test('traverses directory tree up until it finds jest.config', () => {
   expect(summary).toMatchSnapshot();
 });
 
-const jestPath = require.resolve('jest');
-const jestTypesPath = jestPath.replace(/\.js$/, '.d.ts');
-const jestTypesExists = fs.existsSync(jestTypesPath);
+onNodeVersions('<22.6', () => {
+  const jestPath = require.resolve('jest');
+  const jestTypesPath = jestPath.replace(/\.js$/, '.d.ts');
+  const jestTypesExists = fs.existsSync(jestTypesPath);
+
+  (jestTypesExists ? test : test.skip).each([true, false])(
+    'check the config disabled (skip type check: %p)',
+    skipTypeCheck => {
+      writeFiles(DIR, {
+        '__tests__/a-giraffe.js': "test('giraffe', () => expect(1).toBe(1));",
+        'jest.config.ts': `
+        /**@jest-config-loader-options {"transpileOnly":${!!skipTypeCheck}}*/
+        import {Config} from 'jest';
+        const config: Config = { testTimeout: "10000" };
+        export default config;
+      `,
+        'package.json': '{}',
+      });
+
+      const typeErrorString =
+        "TS2322: Type 'string' is not assignable to type 'number'.";
+      const runtimeErrorString = 'Option "testTimeout" must be of type:';
+
+      const {stderr, exitCode} = runJest(DIR, ['-w=1', '--ci=false']);
+
+      if (skipTypeCheck) {
+        expect(stderr).not.toMatch(typeErrorString);
+        expect(stderr).toMatch(runtimeErrorString);
+      } else {
+        expect(stderr).toMatch(typeErrorString);
+        expect(stderr).not.toMatch(runtimeErrorString);
+      }
+
+      expect(exitCode).toBe(1);
+    },
+  );
 
-(jestTypesExists ? test : test.skip).each([true, false])(
-  'check the config disabled (skip type check: %p)',
-  skipTypeCheck => {
+  test('invalid JS in jest.config.ts', () => {
```

Comment:
The diff is a bit weird here - this test is duplicated into a pre-22.6 and post-22.6 version now.